src/execution/execute.js

```diff
@@ -780,9 +823,17 @@ function completeListValue(
   // where the list contains no Promises by avoiding creating another Promise.
   const itemType = returnType.ofType;
   let containsPromise = false;
-  const completedResults = result.map(item => {
-    const completedItem =
-      completeValueCatchingError(exeContext, itemType, fieldASTs, info, item);
+  const completedResults = result.map((item, index) => {
+    const childExePath = exePath.slice();
+    childExePath.push(index);
+
+    const childInfo = Object.assign({}, info, {
```

Comment:
You are right. I was confused by the chain of the execution. Now I see that the `info` object passed down from here is not actually every passed into resolvers. I updated the PR with an appropriate comment.